src/04-modality-specific-files/11-functional-near-infrared-spectroscopy.md

```diff
@@ -0,0 +1,404 @@
+is stored in the field `ShortChannelCount`.
+If the field `ShortChannelCount` is populated, then the optional column `short_channel`
+may be used in `*_channels.tsv` to describe which channels were specified as short.
+
+Generic fields that MUST be present:
+
+{{ MACROS___make_metadata_table(
+   {
+      "TaskName": "REQUIRED",
+   }
+) }}
+
+Generic fields which SHOULD be present: For consistency between studies and institutions,
+we encourage users to extract the values of these fields from the actual raw data.
+Whenever possible, please avoid using ad hoc wording.
+
+{{ MACROS___make_metadata_table(
+   {
+      "InstitutionName": "RECOMMENDED",
+      "InstitutionAddress": "RECOMMENDED",
+      "Manufacturer": "RECOMMENDED",
+      "ManufacturersModelName": "RECOMMENDED",
+      "SoftwareVersions": "RECOMMENDED",
+      "TaskDescription": "RECOMMENDED",
+      "Instructions": "RECOMMENDED",
+      "CogAtlasID": "RECOMMENDED",
+      "CogPOID": "RECOMMENDED",
+      "DeviceSerialNumber": "RECOMMENDED",
+      "RecordingDuration": "RECOMMENDED",
+      "HeadCircumference": "RECOMMENDED",
+      "HardwareFilters": "RECOMMENDED",
+      "SubjectArtefactDescription": "RECOMMENDED",
+   }
+) }}
+
+{{ MACROS___make_metadata_table(
+   {
+      "CapManufacturer": ("RECOMMENDED", "If a custom-made cap is used then the string `custom` should be used. If no cap was used, such as with optodes that are directly taped to the scalp, then the string `none` should be used and the `NIRSPlacementScheme` field may be used to specify the optode placement."),
+      "CapManufacturersModelName": ("RECOMMENDED", "If there is no official model number then description may be provided (for example, `Headband with print (S-M)`). If a cap from a standard manufacturer was modified, then the field should be set to `custom`. If no cap was used, then the `CapManafacturer` field should be `none` and this field should be `n/a`."),
+   }
+) }}
+
+Specific fNIRS fields that MUST be present:
+
+TODO: In the table below the SamplingFrequency data type should be "number or n/a".
+
+{{ MACROS___make_metadata_table(
+   {
+      "SamplingFrequency": ("REQUIRED", "Sampling frequency (in Hz) of all the data in the recording, regardless of their type (for example, `12`).  If individual channels have different sampling rates, then the field here should be specified as `n/a` and the values  should be specified in the `sampling_frequency` column in channels.tsv."),
+      "NIRSChannelCount": "REQUIRED",
+      "NIRSSourceOptodeCount": "REQUIRED",
+      "NIRSDetectorOptodeCount": "REQUIRED",
+      "ACCELChannelCount": "RECOMMENDED but REQUIRED if any channel type is ACC",
+      "GYROChannelCount": "RECOMMENDED but REQUIRED if any channel type is GYRO",
+      "MAGNChannelCount": "RECOMMENDED but REQUIRED if any channel type is MAGN",
+   }
+) }}
+
+Specific fNIRS fields that SHOULD be present:
+
+TODO: The SourceType and DetectorType data types should be "string or mixed".
```

Comment:
Simillarly @sappelhoff , how can I add `"mixed"` to the data type for the source and detector? I tried editing the metadata.yaml but I couldnt make it work.

Reply:
Of course! "mixed" is a string, so this is fine as it is. My bad, I was too in the weeds and missed the bigger picture. Thanks @sappelhoff